src/m365/pp/commands/website/website-webrole-remove.ts

```diff
@@ -0,0 +1,192 @@
+        asAdmin: !!args.options.asAdmin,
+        force: !!args.options.force
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-e, --environmentName <environmentName>'
+      },
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '--websiteId [websiteId]'
+      },
+      {
+        option: '--websiteName [websiteName]'
+      },
+      {
+        option: '--asAdmin'
+      },
+      {
+        option: '-f, --force'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['websiteId', 'websiteName'] }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+        if (args.options.websiteId && !validation.isValidGuid(args.options.websiteId)) {
+          return `${args.options.websiteId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: any): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Removing websrole '${args.options.id}'...`);
+    }
+
+    if (args.options.force) {
+      await this.deleteWebSiteWebRole(args);
+    }
+    else {
+      const result = await cli.promptForConfirmation({ message: `Are you sure you want to remove webrole '${args.options.id}'?` });
+
+      if (result) {
+        await this.deleteWebSiteWebRole(args);
+      }
+    }
+  }
+
+  private async getWebSiteId(dynamicsApiUrl: string, args: CommandArgs): Promise<any> {
+    if (args.options.websiteId) {
+      return args.options.websiteId;
+    }
+
+    const options: PpWebSiteOptions = {
+      environmentName: args.options.environmentName,
+      id: args.options.websiteId,
+      name: args.options.websiteName,
+      output: 'json'
+    };
+
+    const requestOptions: CliRequestOptions = {
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    if (options.name) {
+      requestOptions.url = `${dynamicsApiUrl}/api/data/v9.2/powerpagesites?$filter=name eq '${options.name}'&$select=powerpagesiteid`;
+      const result = await request.get<{ value: any[] }>(requestOptions);
+
+      if (result.value.length === 0) {
+        throw `The specified website '${args.options.websiteName}' does not exist.`;
+      }
+      return result.value[0].powerpagesiteid;
+    }
+
+  }
+
+  private async getWebSiteWebRoleId(dynamicsApiUrl: string, args: CommandArgs): Promise<any> {
+
+    const requestOptions: CliRequestOptions = {
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const websiteId = await this.getWebSiteId(dynamicsApiUrl, args);
+
+    if (websiteId) {
```

Comment:
This if-statement is superfluous i believe. websiteId should not be null. If it is null, an exception would have been thrown earlier, about the site id not existing.